lm_eval/models/mlc_llm.py

```diff
@@ -222,4 +223,174 @@ def greedy_until(
             )
             print(f"\r{num}/{len(requests)} requests processed", end="")
 
-        return results
\ No newline at end of file
+        return results
+
+
+class MLCServe(BaseLM):
+    def __init__(
+        self,
+        model_name: str,
+        ip: str = "0.0.0.0",
+        port: int = 32777,
+        batch_size: int = 1,
+        temperature: float = 0.0,
+        top_p: float = 1.0,
+        parallel: bool = False,
+    ):
+        super().__init__()
+
+        self.model_name = model_name
+        self.ip = ip
+        self.port = port
+        self._batch_size = batch_size
+        self.temperature = temperature
+        self.top_p = top_p
+
+        self.url_suffix = "/v1/chat/completions"
+        self.parallel = parallel or batch_size > 1
+        assert batch_size == 1 and not parallel, "Please insert batch size bigger than 1 for parallel regime"
+
+        token = os.environ["OCTOAI_TOKEN"]
+        self.headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {token}",
+        }
+
+    @property
+    def eot_token_id(self):
+        return self.tokenizer.eos_token_id
+
+    @property
+    def max_length(self):
+        return 2048
+
+    @property
+    def max_gen_toks(self):
+        return 256
+
+    @property
+    def batch_size(self):
+        return self._batch_size
+
+    @property
+    def device(self):
+        return self._device
+
+    def tok_encode(self, string: str):
+        raise NotImplementedError("Cannot call tokenizer by API")
+
+    def tok_decode(self, tokens: Iterable[int]):
+        raise NotImplementedError("Cannot call tokenizer by API")
+
+    def create_chat_completion_payload(
+        self,
+        prompt,
+        stop_token = None,
+    ):
+        payload = {
+            "model": self.model_name,
+            "messages": [
+                {
+                    "role": "user",
+                    "content": prompt,
+                }
+            ],
+            "stream": False,
+            "stop": [stop_token],
```

Comment:
`stop_token` is already a list, so there will be error when validating request by pydantic